apm-agent-plugins/apm-log-correlation-plugin/src/main/java/co/elastic/apm/agent/slf4j/Slf4JMdcActivationListener.java

```diff
@@ -41,44 +41,101 @@
 
 public class Slf4JMdcActivationListener implements ActivationListener {
 
-    // the string concatenation prevents the shade plugin from relocating org.slf4j.MDC to co.elastic.apm.agent.shaded.slf4j.MDC
-    // the toString prevents constant folding, which would also make the shade plugin relocate
-    private static final String ORG_SLF4J_MDC = "org." + "slf4j.MDC".toString();
+    // prevents the shade plugin from relocating org.slf4j.MDC to co.elastic.apm.agent.shaded.slf4j.MDC
+    private static final String SLF4J_MDC = "org!slf4j!MDC".replace('!', '.');
+    private static final String LOG4J_MDC = "org.apache.log4j.MDC";
+    private static final String LOG4J2_MDC = "org.apache.logging.log4j.ThreadContext";
+
     private static final String TRACE_ID = "trace.id";
     private static final String TRANSACTION_ID = "transaction.id";
     private static final Logger logger = LoggerFactory.getLogger(Slf4JMdcActivationListener.class);
 
     // Never invoked- only used for caching ClassLoaders that can't load the slf4j MDC class
     private static final MethodHandle NOOP = MethodHandles.constant(String.class, "ClassLoader cannot load slf4j API");
```

Comment:
The returned message is not really used, but change it to something generic so that it won't be misleading (same for the comment)